### PR TITLE
[FLINK-12438][doc-zh]Translate Task Lifecycle document into Chinese

### PR DESCRIPTION
**Pull request by hapihu**
Translate Task Lifecycle document into Chinese

### CODE REVIEW

**Comment by wuchong** [comment]
cc @RocMarshal , do you want to help reviewing this page?

**Comment by RocMarshal** [comment]
Good job. @hapihu . And thanks @wuchong for assigning. I'll review this PR. Wait a moment, please.

**Review by RocMarshal — COMMENTED** [review]
Thank @hapihu you for your contribution. And I left some comments before line 78 in the translated page . Please let me know what do you think of it.

File: docs/content.zh/docs/internals/task_lifecycle.md [hunk 1]
```diff
@@ -24,181 +24,123 @@ specific language governing permissions and limitations
...
+<a name='task-lifecycle'> </a>
+
 # Task 生命周期
 
-A task in Flink is the basic unit of execution. It is the place where each parallel instance of an
-operator is executed. As an example, an operator with a parallelism of *5* will have each of its
-instances executed by a separate task.
+Task 是 Flink 的基本执行单元。算子的每个并行实例都在 task 里执行。例如，一个并行度为 5 的算子，它的每个实例都由一个单独的 task 来执行。
+
+在 Flink 流式计算引擎里，`StreamTask` 是所有不同 task 子类的基础。本文会深入讲解  `StreamTask` 生命周期的不同阶段，并描述每个阶段的主要方法。
```
[the remark this answers is not in the document]
Comment (hapihu) [reply docs/content.zh/docs/internals/task_lifecycle.md:33]: Well, I think your translation reads more smoothly.
I made a slight change, and the result looks like this：

> `StreamTask` 是 Flink 流式计算引擎中所有不同 task 子类的基础。本文会深入讲解  `StreamTask` 生命周期的不同阶段，并阐述每个阶段的主要方法。

What do you think of it?

File: docs/content.zh/docs/internals/task_lifecycle.md [hunk 2]
```diff
@@ -24,181 +24,123 @@ specific language governing permissions and limitations
 under the License.
 -->
 
+<a name='task-lifecycle'> </a>
+
```
[the remark this answers is not in the document]
Comment (hapihu) [reply docs/content.zh/docs/internals/task_lifecycle.md:28]: OK, I get it.

File: docs/content.zh/docs/internals/task_lifecycle.md [hunk 3]
```diff
@@ -24,181 +24,123 @@ specific language governing permissions and limitations
...
+        // 初始化阶段
         OPERATOR::setup
             UDF::setRuntimeContext
         OPERATOR::initializeState
         OPERATOR::open
             UDF::open
-
-        // processing phase (called on every element/watermark)
+
```
[the remark this answers is not in the document]
Comment (hapihu) [reply docs/content.zh/docs/internals/task_lifecycle.md:50]: I think it's more consistent to have an empty line here

**Comment by hapihu** [comment]
Hello, @RocMarshal, I made some changes in response to your suggestion. plz help me review again~ 
Thank you very much!

File: docs/content.zh/docs/internals/task_lifecycle.md [hunk 4]
```diff
@@ -24,181 +24,119 @@ specific language governing permissions and limitations
...
 
-In the case of running till completion, *i.e.* there is no more input data to process, after exiting from the `run()` 
-method, the task enters its shutdown process. Initially, the timer service stops registering any new timers (*e.g.* from 
-fired timers that are being executed), clears all not-yet-started timers, and awaits the completion of currently 
-executing timers. Then the `finishAllOperators()` notifies the operators involved in the computation by
-calling the `finish()` method of each operator. Then, any buffered output data is flushed so that they can be processed
-by the downstream tasks, and finally the task tries to clear all the resources held by the operators by calling the
-`close()` method of each one. When opening the different operators, we mentioned that the order is from the
-last to the first. Closing happens in the opposite manner, from first to last.
+在运行到完成的情况下，即没有更多的输入数据要处理，从run()方法退出后，task 进入关闭阶段。首先定时器服务停止注册任何新的定时器（比如从正在执行的定时器里注册），清理掉所有还未启动的定时器，并等待当前执行中的定时器运行结束。然后 `finishAllOperators()` 方法通过调用每个算子的 `finish()` 方法来通知所有参与计算的算子。然后所有缓存的输出数据会刷出去以便下游 task 处理，最终 task 通过调用每个算子的 `close()` 方法来尝试清理掉算子持有的所有资源。之前我们提到不同算子开启时，是从后往前依次调用；关闭时刚好相反，从前往后依次调用。
```
Comment (RocMarshal) [inline docs/content.zh/docs/internals/task_lifecycle.md:121]:
```suggestion
在运行到完成的情况下，即没有更多的输入数据要处理，从run()方法退出后，task 进入关闭阶段。首先定时器服务停止注册任何新的定时器（比如从正在执行的定时器里注册），清理掉所有还未启动的定时器，并等待当前执行中的定时器运行结束。然后通过调用 `finishAllOperators()` 方法调用每个算子的 `finish()` 方法来通知所有参与计算的算子。然后所有缓存的输出数据会刷出去以便下游 task 处理，最终 task 通过调用每个算子的 `close()` 方法来尝试清理掉算子持有的所有资源。与我们之前提到的开启算子不同是，开启时从后往前依次调用 `open()`；而关闭时刚好相反，从前往后依次调用 `close()`。
```

File: docs/content.zh/docs/internals/task_lifecycle.md [hunk 5]
```diff
@@ -24,181 +24,119 @@ specific language governing permissions and limitations
...
-
-### Interrupted Execution
-
-In the previous sections we described the lifecycle of a task that runs till completion. In case the task is cancelled 
-at any point, then the normal execution is interrupted and the only operations performed from that point on are the timer 
-service shutdown, the task-specific cleanup, the closing of the operators, and the general task cleanup, as described
-above.
+最后，当所有算子都已经关闭，所有资源都已被释放时，task 关掉它的定时器服务，进行特定 task 的清理操作，例如清理掉所有内部缓存，然后进行常规的 task 清理操作，包括关闭所有的输出管道，清理所有输出缓存等。
+
+**Checkpoints:** 之前我们看到在执行 `initializeState()` 方法期间，在从异常失败中恢复的情况下，task 和它内部的所有算子函数都从最后一次成功的 checkpoint 数据里获取对应的状态信息。Flink 里的 checkpoint 是根据用户自定义的时间间隔周期执行的，并且在一个与主 task 线程不同的单独线程里执行。这也是我们没有把 checkpoint 过程涵盖在 task 生命周期的主要阶段里的原因。简而言之，Flink 作业的输入数据 source task 会定时插入一种叫 `checkpoint barrier` 的特殊数据，并跟正常数据一起从 source 流入到 sink。source task 在处于运行模式后发送这些  barrier  ，同时会假设 `CheckpointCoordinator` 也在运行。当 task 接收到这样的 barrier 之后，会通过 task 算子里的 `snapshotState()` 方法调度 checkpoint 线程执行具体任务。在 checkpoint 处理期间，task 依然可以接收输入数据，但是数据会被缓存起来，当 checkpoint 执行成功之后才会被处理和发送到下游算子。
```
Comment (RocMarshal) [inline docs/content.zh/docs/internals/task_lifecycle.md:131]:
```suggestion
**Checkpoints:** 之前我们看到在执行 `initializeState()` 方法期间，在从异常失败中恢复的情况下，task 和它内部的所有算子函数都从最后一次成功的 checkpoint 数据里获取对应的状态信息。Flink 里的 checkpoint 是根据用户自定义的时间间隔周期执行的，并且在一个与主 task 线程不同的单独线程里执行。这也是我们没有把 checkpoint 过程涵盖在 task 生命周期的主要阶段里的原因。简而言之，Flink 作业的输入数据 source task 会定时插入一种叫 `checkpoint barrier` 的特殊数据，并跟正常数据一起从 source 流入到 sink。source task 在处于运行模式后发送这些  barrier，同时会假设 `CheckpointCoordinator` 也在运行。当 task 接收到这样的 barrier 之后，会通过 task 算子里的 `snapshotState()` 方法调度 checkpoint 线程执行具体任务。在 checkpoint 处理期间，task 依然可以接收输入数据，但是数据会被缓存起来，当 checkpoint 执行成功之后才会被处理和发送到下游算子。 
```

**Review by RocMarshal — COMMENTED** [review]
@hapihu I left some comments after line `70` in the translated page.
Please let me know what do you think of it . Thank you.

File: docs/content.zh/docs/internals/task_lifecycle.md [hunk 6]
```diff
@@ -24,181 +24,119 @@ specific language governing permissions and limitations
...
+	   	    open-operators
+		    run
+		    finish-operators
+		    close-operators
+		    task-specific-cleanup
+		    common-cleanup
+
+如上所示，在恢复 task 配置和初始化一些重要的运行时参数之后，task 的下一步是读取 task 级别的初始状态数据。这一步在 `setInitialState()` 方法里完成，在下面两个场景特别重要：
+
+1. 当 Task 从失败中恢复，从最后一次成功的 checkpoint 重启的时候
```
[the remark this answers is not in the document]
Comment (hapihu) [reply docs/content.zh/docs/internals/task_lifecycle.md:102]: ok, I made a slight change, and the result looks like this：
>  1. 当 Task 从失败中恢复并从最近一次成功的 checkpoint 重启的时候

Change  `、` to `.`.
What do you think of it?

File: docs/content.zh/docs/internals/task_lifecycle.md [hunk 7]
```diff
@@ -24,181 +24,119 @@ specific language governing permissions and limitations
...
+2. 当 Task 从 [savepoint]({{< ref "docs/ops/state/savepoints" >}}) 里恢复的时候。
```
[the remark this answers is not in the document]
Comment (hapihu) [reply docs/content.zh/docs/internals/task_lifecycle.md:103]: ok, I made a slight change, and the result looks like this：

> 2. 当 Task 从 [savepoint]({{< ref "docs/ops/state/savepoints" >}}) 恢复的时候。
Change `、` to `.`.
What do you think of it?

File: docs/content.zh/docs/internals/task_lifecycle.md [hunk 8]
```diff
@@ -24,181 +24,119 @@ specific language governing permissions and limitations
...
+
+**Checkpoints:** 算子的 `snapshotState()` 方法是在收到 checkpoint barrier 后异步调用的。Checkpoint 在处理阶段执行，即算子打开之后，结束之前的这个阶段。这个方法的职责是存储算子的当前状态到一个特定的[状态后端]({{< ref "docs/ops/state/state_backends" >}})，当作业失败后恢复执行时会从这个后端恢复状态数据。下面我们简要描述了 Flink 的 checkpoint 机制，如果想了解更多 Flink checkpoint 相关的原理，可以读一读 [数据流容错]({{< ref "docs/learn-flink/fault_tolerance" >}})。
+
+<a name="task-lifecycle-1"> </a>
+
+## Task 生命周期
+
+在上文对算子主要阶段的简要介绍之后，本节将详细介绍 task 在集群执行期间是如何调用相关方法的。这里所说的阶段主要包含在 `StreamTask` 类的 `invoke()` 方法里。本文档的剩余部分会分成两个子章节，其中一节描述了 task 在常规没有失败情况下执行的阶段（请参考 [常规执行](#normal-execution)），另外一节描述了 task 取消之后执行的阶段（请参考 [中断执行](#interrupted-execution)），不管是手动取消还是其他原因（比如执行期间遇到异常）导致的取消。
+
+<a name="normal-execution"> </a>
```
[the remark this answers is not in the document]
Comment (hapihu) [reply docs/content.zh/docs/internals/task_lifecycle.md:81]: ok

**Comment by hapihu** [comment]
Hello, @RocMarshal, 
Excuse me for taking up your time.
I have finished the modification according to your suggestion. Would you like to review it for me again.
Thank you very much!

**Review by RocMarshal — COMMENTED** [review]
Thanks @hapihu for the update. LGTM+1.
Now, ping @wuchong .

**Comment by hapihu** [comment]
@wuchong 
Excuse me for taking up your time.
Can you help me to review it again? thanks

**Pull request merged**